foundry.md

```diff
@@ -0,0 +1,294 @@
+                 ...
+        </instrs>
+      [owise]
+
+```
+
+### Get/set storage
+
+```k
+    rule [hostCall-getStorage]:
+        <instrs> hostCall ( "env" , "getStorage" , [ i32  i32  i32  .ValTypes ] -> [ .ValTypes ] )
+              => foundryGetStorage( getBuffer(OWNER_HANDLE), getBuffer(KEY_HANDLE), DEST_HANDLE)
+                 ...
+        </instrs>
+        <locals>
+          0 |-> <i32> OWNER_HANDLE
+          1 |-> <i32> KEY_HANDLE
+          2 |-> <i32> DEST_HANDLE
+        </locals>
+        <callee> #foundryRunner </callee>
+
+    syntax InternalInstr ::= foundryGetStorage(BytesResult, BytesResult, Int)
+ // -------------------------------------------------------------------------
+    rule [testapi-getStorage]:
+        <instrs> foundryGetStorage(OWNER:Bytes, KEY:Bytes, DEST_HANDLE)
+              => #setBuffer(DEST_HANDLE, VAL)
+                 ...
+        </instrs>
+        <account>
+          <address> OWNER </address>
+          <storage> ...  wrap(KEY) Bytes2Bytes|-> wrap(VAL) ... </storage>
+          ...
+        </account>
+
+    rule [testapi-getStorage-err]:
+        <instrs> foundryGetStorage(_, _, DEST_HANDLE)
+              => #setBuffer(DEST_HANDLE, .Bytes)
+                 ...
+        </instrs>
+      [owise]
+
+```
+
+### Assertions/assumptions
+
+```k
+    rule [hostCall-assertBool]:
+        <instrs> hostCall ( "env" , "assertBool" , [ i32 .ValTypes ] -> [ .ValTypes ] )
+              => #if P =/=Int 0
+                 #then .K
+                 #else #throwException(ExecutionFailed, "assertion failed")
+                 #fi
+                 ...
+        </instrs>
+        <locals>
+          0 |-> <i32> P
+        </locals>
+
+
+    rule [hostCall-assumeBool]:
+        <instrs> hostCall ( "env" , "assumeBool" , [ i32 .ValTypes ] -> [ .ValTypes ] )
+              => #assume(P) ...
+        </instrs>
+        <locals>
+          0 |-> <i32> P
+        </locals>
+
+    syntax IternalInstr ::= #assume(Int)
+ // ------------------------------------
+    rule [assume]:
+        <instrs> #assume(P) => . ... </instrs>
+      ensures P =/=Int 0
+
+```
+
+### Prank
+
+```k
+    rule [hostCall-startPrank]:
+        <instrs> hostCall ( "env" , "startPrank" , [ i32  .ValTypes ] -> [ .ValTypes ] )
+              => #startPrank(getBuffer(ADDR_HANDLE)) ...
+        </instrs>
+        <locals>
+          0 |-> <i32> ADDR_HANDLE
+        </locals>
+        <callee> #foundryRunner </callee>
+
+    syntax InternalInstr ::= #startPrank(BytesResult)
+ // -------------------------------------------------
+    rule [startPrank]:
+        <instrs> #startPrank(ADDR:Bytes) => . ... </instrs>
+        <callee> #foundryRunner => ADDR </callee>
+        <prank> false => true </prank>
+
+    rule [startPrank-not-allowed]:
+        <instrs> #startPrank(_:Bytes)
+              => #throwException(ExecutionFailed, "Only the test contract can start a prank") 
```

Comment:
Should the message also say "and the test contract can't start a prank while already pranking"?